From: Alexander Graf <agraf@suse.de>
Date: Weg, 20 Mar 2013 15:17:25 +0100
Subject: [PATCH] ARM: kvm: fix IOMMU dependency check
Patch-Mainline: No

ARM KVM does not use the KVM internal IOMMU framework. In fact,
nobody should use it except for legacy x86. So make sure that
only x86 gets the legacy iommu callbacks defined.

Signed-off-by: Alexander Graf <agraf@suse.de>

Index: linux-3.9-rc3-master/include/linux/kvm_host.h
===================================================================
--- linux-3.9-rc3-master.orig/include/linux/kvm_host.h
+++ linux-3.9-rc3-master/include/linux/kvm_host.h
@@ -705,7 +705,7 @@ void kvm_free_irq_source_id(struct kvm *
 /* For vcpu->arch.iommu_flags */
 #define KVM_IOMMU_CACHE_COHERENCY	0x1
 
-#ifdef CONFIG_IOMMU_API
+#if defined(CONFIG_IOMMU_API) && (defined(__i386__) || defined(__x86_64__))
 int kvm_iommu_map_pages(struct kvm *kvm, struct kvm_memory_slot *slot);
 void kvm_iommu_unmap_pages(struct kvm *kvm, struct kvm_memory_slot *slot);
 int kvm_iommu_map_guest(struct kvm *kvm);
